src/network/networkctl.c

```diff
@@ -2919,6 +2930,388 @@ static int verb_reconfigure(int argc, char *argv[], void *userdata) {
+                                return r;
+
+                        r = add_config_to_edit(&context, path, dropins);
+                        if (r < 0)
+                                return r;
+
+                        /* We only operate on .network file here, so no need to check or trigger udevd */
+                        networkd_reload = true;
+
+                        continue;
+                }
+
+                if (ENDSWITH_SET(*name, ".network", ".netdev"))
+                        networkd_reload = true;
+                else if (endswith(*name, ".link"))
+                        udevd_trigger = true;
+                else
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Invalid network config name '%s'.", *name);
+
+                r = get_config_files_by_name(*name, &path, &dropins);
+                if (r == -ENOENT) {
+                        if (arg_drop_in)
+                                return log_error_errno(r, "Cannot find network config '%s'.", *name);
+
+                        log_debug("No existing network config '%s' found, creating a new file.", *name);
+
+                        path = path_join(NETWORK_DIRS[0], *name);
+                        if (!path)
+                                return log_oom();
+
+                        r = edit_files_add(&context, path, NULL, NULL);
+                        if (r < 0)
+                                return r;
+                        continue;
+                }
+                if (r < 0)
+                        return log_error_errno(r, "Failed to get the path of network config '%s': %m", *name);
+
+                r = add_config_to_edit(&context, path, dropins);
+                if (r < 0)
+                        return r;
+        }
+
+        r = do_edit_files_and_install(&context);
+        if (r < 0)
+                return r;
+
+        if (arg_no_reload)
+                return 0;
+
+        if (!sd_booted() || running_in_chroot() > 0) {
+                log_debug("System is not booted with systemd or is running in chroot, skipping reload.");
+                return 0;
+        }
+
+        _cleanup_(sd_bus_flush_close_unrefp) sd_bus *bus = NULL;
+
+        r = sd_bus_open_system(&bus);
+        if (r < 0)
+                return log_error_errno(r, "Failed to connect to system bus: %m");
+
+        if (udevd_trigger) {
+                r = udevd_reload(bus);
+                if (r < 0)
+                        return r;
+                if (r > 0) {
+                        r = safe_fork("(udevadm)",
+                                      FORK_RESET_SIGNALS|FORK_CLOSE_ALL_FDS|FORK_DEATHSIG|FORK_LOG|FORK_WAIT|FORK_FLUSH_STDIO,
+                                      NULL);
+                        if (r < 0)
+                                return r;
+                        if (r == 0) { /* Child */
+                                execl("/usr/bin/udevadm", "udevadm", "trigger", "--subsystem-match=net",
```

Comment:
Just use `execlp` and don't specify the path to `udevadm`. No reason not to use `$PATH`.